cdap-ui/app/cdap/components/DataPrep/store/DataPrepActionCreator.js

```diff
@@ -120,3 +123,52 @@ export function setWorkspace(workspaceId) {
       });
   });
 }
+
+export function getWorkspaceList(workspaceId) {
+  let namespace = NamespaceStore.getState().selectedNamespace;
+
+  return Rx.Observable.create((observer) => {
```

Comment:
If we are not going to subscribe to this action then it would be helpful if we return an observable. Since response for the action is always going to update the store could probably just not return anything. Right now because of this we have `getWorkspaceList().subscribe()` in all usages.